doc/superbuild/source/developers/style_guide/SourceCode.rst

```diff
@@ -395,46 +338,81 @@ Coding Conventions and Rules
 Formatting
 ----------
 
-All new code added to SUNDIALS should be linted with  `clang-tidy
-<https://clang.llvm.org/extra/clang-tidy/>`_ and formatted with `clang-format
-<https://clang.llvm.org/docs/ClangFormat.html>`_. The ``.clang-tidy`` and
+All new code added to SUNDIALS should be formatted with `clang-format
+<https://clang.llvm.org/docs/ClangFormat.html>`_. The 
 ``.clang-format`` files in the root of the project define our configurations
 for the tools respectively. To run both tools on a directory, you can run
 
-.. code-block::
+If clang-format breaks lines in a way that is unreadable, use ``//`` to break the line. For example,
+sometimes (mostly in C++ code) you may have code like this:
 
-   cd scripts
-   ./tidy_and_format.sh <path to directory to tidy and format> <path to builddir>
+.. code-block:: cpp
 
-For clang-tidy to work, the build directory must have a ``compile_commands.json``, which 
-can be generated by setting the CMake option ``CMAKE_EXPORT_COMPILE_COMMANDS``
-to ``ON``. If you just want to run the formatter, then you can run
+   MyClass::callAFunctionOfSorts::doSomething().doAnotherThing().doSomethingElse();
 
-.. code-block::
+That you would like to format as (for readability):
 
-   cd scripts
-   ./format.sh <path to directory to format>
+.. code-block:: cpp
+
+   MyObject::callAFunctionOfSorts()
+         .doSomething()
+         .doAnotherThing()
+         .doSomethingElse();
 
-It may be necessary to override clang-tidy at times. This can be done with the
-``NOLINT`` magic comments e.g.,
+Clang-format might produce something like:
 
 .. code-block:: cpp
 
-  template<class GkoSolverType, class GkoMatrixType>
-  int SUNLinSolFree_Ginkgo(SUNLinearSolver S)
-  {
-    auto solver{static_cast<LinearSolver<GkoSolverType, GkoMatrixType>*>(S->content)};
-    delete solver; // NOLINT
-    return SUNLS_SUCCESS;
-  }
-
-  class BaseObject {
-  protected:
-    // NOLINTNEXTLINE(cppcoreguidelines-non-private-member-variables-in-classes)
-    SUNContext sunctx_{};
-  };
-
-See the clang-tidy documentation for more details.
+   MyObject::callAFunctionOfSorts().doSomething().doAnotherThing()
+         .doSomethingElse();
+```
+
+unless you add the `//`.
+
+.. code-block:: cpp
+
+   MyObject::callAFunctionOfSorts()
+         .doSomething()       //
+         .doAnotherThing()    //
+         .doSomethingElse();  //
+
+There are other scenarios (e.g., a function call with a lot of parameters) where doing this type of line break is useful for readability too.
+
+..
```

Comment:
Did you intend for the text below to be commented out?

Reply:
Yes, I did not want to lose what I had wrote but I also did not want it visible in the developer docs at this moment.